### ISSUE TITLE
Unable to open NVDA settings dialog

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#9564]
### Steps to reproduce:
1. Open NVDA and open its log viewer
2. Open the save as dialog
3. Press ctrl+nvda+g
4. Close the save as dialog with escape and close the log viewer
5. Press ctrl+nvda+g
6. Got to the nvda menu and open the settings by pressing enter on the menu item
7. Press ctrl+nvda+s

### Actual behavior:
After doing steps 1 through 4 you note that NVDA does not open the general settings dialog. At least it does not appear on the screen. However, the dialog seems open because pressing the key stroke again displays a dialog which says that a dialog is already open (step 5). Moreover, it is not possible to open the settings dialog in the NVDA menu (step 6). Step 7 works though. That means the settings cathegories can only be accessed by pressing their shortcuts.

### Expected behavior:
NVDA should not fire the general settings dialog when doing steps 1 through 4 and should allow the user to access the settings pannel afterwards.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
NVDA 2019.1.1

#### Windows version:
Windows 10 1809 Update

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

**Comment by Adriani90** [comment]
This might also be related to #3361. However, in this case there is no error or any warning when pressing ctrl+nvda+g the first time.

**Comment by Qchristensen** [comment]
I might have got this wrong, however, what happens for me seems to be exactly what is supposed to happen:

1. Open NVDA and open its log viewer
(log viewer opens)
2. Open the save as dialog
(save as dialog opens)
3. Press ctrl+nvda+g
(General settings dialog opens)
4. Close the save as dialog with escape and close the log viewer
(alt+tab to save as dialog?  or did you want me to close the general settings dialog first?  I've alt+tabbed)
5. Press ctrl+nvda+g
(I get an error that an NVDA dialog is already open - the general settings dialog I alt+tabbed away from in the previous step)
6. Got to the nvda menu and open the settings by pressing enter on the menu item
(As per step 5, I get the error as the settings dialog is still open)
7. Press ctrl+nvda+s
(Synthesizer dialog opens).

The synthesizer dialog needs to be able to open while another NVDA dialog is open because it is setup so you can open it from the Speech settings category of the settings dialog.

If, in step 7 (or anywhere after you've already opened the settings dialog), you press say NVDA+control+k to open the keyboard settings, you'll get the error because the dialog is already open (my personal preference would be, where the other dialog is the settings dialog, to return to it, and go to the category that opens with that keystroke, but that is perhaps a different issue).

The issue in #3361 still occurs (if you open the log viewer save as, then NVDA settings, you can't navigate with TAB), but I can still open the settings dialog,

**Comment by Brian1Gaff** [comment]
One other dialogue funny is the issue. You poll for an update and one is 
found, you get it to download it, many times when it returns the dialogue to 
update is still active and one can download it again. What should occur is 
that always the ready to install dialogue should appear and the download 
should close.
 I've noticed this happen a lot on Alpha snaps, and of course I cannot test 
it on the stable branch.

Could it all be due to either some signal to close or open being missed 
somewhere making things more erratic than they should be?

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
-----

**Comment by DrSooom** [comment]
@Brian1Gaff: Could you please open a separate issue regarding the update dialog, as it seems that it has nothing to do with this issue here? Thanks.

@Adriani90: "But" insteadd of "Bug" in the issue title.

**Comment by Adriani90** [comment]
@Brian1Gaff your issue is covered in #4579.

**Comment by Adriani90** [comment]
@Qchristensen I am testing again with NVDA 2019.3 Beta 1, In my case, step 3 still does not open the dialog for general settings. NVDA does not report anything.
Afterwards, Except for NVDA+ctrl+s, all other attempts to open other settings cathegories result in the notification which says that a settings dialog is already open.
It is however strange to me that only opening the synthesizer dialog works though.
cc: @feerrenrut, @leonardder

**Comment by Qchristensen** [comment]
The synthesizer dialog is a separate dialog to the settings dialog, and because it can be opened via the "Change" button on the settings dialog, both need to be able to be open at the same time.

I just tried this again with 2019.3 beta 1 and it still works as I noted previously in [this comment](https://github.com/nvaccess/nvda/issues/9564#issuecomment-492452330)

**Comment by CyrilleB79** [comment]
I cannot reproduce either on NVDA 2019.3beta1.
@Adriani90, could you provide a log so that we can double check that we all execute the same steps?
Specifically, you did not comment to confirm or infirm @Qchristensen's following statement:

> (alt+tab to save as dialog? or did you want me to close the general settings dialog first? I've alt+tabbed)

**Comment by Adriani90** [comment]
@Qchristensen your steps are leading to my expected behavior. However, it seems in step 3 you get another result than me. IN my case, the general settings dialog just does not open. I don't really know what I should do, I also tried with disabled addons and the dialog still does not open although I get the error saying that a dialog is open.

**Comment by Adriani90** [comment]
Hmm very strange, I have created a portable version of NVDA without copying my settings and the issue is not occuring. aFter that, I have started the installed version again and now the issue is not occuring anymore. Very strange...
Anyway, closing as works for me :)
